cobalt/watchdog/instrumentation_log.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/watchdog/instrumentation_log.h"
+
+#include <string>
+#include <vector>
+
+namespace cobalt {
+namespace watchdog {
+
+void InstrumentationLog::LogEvent(std::string event) {
+  starboard::ScopedLock scoped_lock(buffer_mutex_);
+  if (buffer_.CurrentIndex() == 0 || event != **(buffer_.End())) {
```

Comment:
Why this check for End()? Is the intent not to reinsert the same value twice back to back? When could that happen?

Reply:
yes, that's the intent. Simply calling the same function 100 times in a row would result in the 100 identical frames. a